### PR TITLE
Added Person schema and updated Post to include block array

### PR DESCRIPTION
**Pull request by madisonbullard**
Added some more type helpers and updated queries to reflect changes to add a Person schema and update Post to contain an array of Blocks.

### CODE REVIEW

File: api/post/types.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    author->
+  }
+`
+export const postsQuery = `*[_type == "post"]${postProjection}`
+
+const PostPick = Post.pick({ _id: true, title: true, body: true })
+
+const PostRes = PostPick.merge(z.object({ author: Person }))
+export const Posts = z.array(PostRes)
+export type TPost = z.infer<typeof PostRes>
```
Comment (madisonbullard) [inline api/post/types.ts:36]: Reeeeally wanted to colocate the query and the type data of the expected query response, despite `postsQuery` not really fitting the `/types.ts` filename.

We might find that moving to/building a typed query builder would yield an easier-to-grok interface, but I'd prefer to wait for that to become obvious before introducing that complexity.

File: api/sanity/array/types.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { z } from 'zod'
+
+export function sanityArrayOf<T extends z.ZodTypeAny>(type: T) {
+  return z.object({
+    style: z.string(),
+    _key: z.string(),
+    markDefs: z.array(z.unknown()),
```
Comment (madisonbullard) [inline api/sanity/array/types.ts:7]: We'll need to update this once we figure out what kind of data goes here. Currently just see an empty array, thus the `z.unknown()`

File: api/sanity/block/types.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { z } from 'zod'
+
+export const SanityBlock = z.object({
+  marks: z.array(z.unknown()),
```
Comment (madisonbullard) [inline api/sanity/block/types.ts:4]: Same here

Comment (daltn) [reply api/sanity/block/types.ts:4]: thanks for noting

**Review by daltn — APPROVED** [review]

**Pull request merged**